app/src/main/java/org/zalando/nakadi/config/WebConfig.java

```diff
@@ -53,20 +75,44 @@ public TimeoutCallableProcessingInterceptor timeoutInterceptor() {
         return new TimeoutCallableProcessingInterceptor();
     }
 
+    @Bean
+    public FilterRegistrationBean requestRejectedFilter() {
+        return createFilterRegistrationBean(new RequestRejectedFilter(), Ordered.HIGHEST_PRECEDENCE);
+    }
+
     @Bean
     public FilterRegistrationBean flowIdRequestFilter() {
-        return createFilterRegistrationBean(new FlowIdRequestFilter(), Ordered.HIGHEST_PRECEDENCE + 1);
+        return createFilterRegistrationBean(new FlowIdRequestFilter(), Ordered.HIGHEST_PRECEDENCE + 10);
     }
 
     @Bean
-    public FilterRegistrationBean traceRequestFilter() {
-        return createFilterRegistrationBean(new TracingFilter(authorizationService), Ordered.LOWEST_PRECEDENCE - 1);
+    public FilterRegistrationBean loggingFilter() {
+        return createFilterRegistrationBean(
+                new LoggingFilter(nakadiKpiPublisher, authorizationService, featureToggleService, accessLogEventType),
+                Ordered.HIGHEST_PRECEDENCE + 20);
+    }
+
+    @Bean
+    public FilterRegistrationBean monitoringRequestFilter() {
+        return createFilterRegistrationBean(
+                new MonitoringRequestFilter(metricRegistry, perPathMetricRegistry, authorizationService),
+                Ordered.HIGHEST_PRECEDENCE + 30);
     }
 
     @Bean
     public FilterRegistrationBean gzipBodyRequestFilter(final ObjectMapper mapper) {
+        return createFilterRegistrationBean(new GzipBodyRequestFilter(mapper), Ordered.HIGHEST_PRECEDENCE + 40);
+    }
+
+    @Bean
+    public FilterRegistrationBean traceRequestFilter() {
+        return createFilterRegistrationBean(new TracingFilter(), Ordered.HIGHEST_PRECEDENCE + 50);
```

Comment:
Seems to be better to have the following order: 
1. TracingFilter
2. MonitoringRequestFilter
3. FlowIdRequestFilter
4. LoggingFilter
5. RequestRejectedFilter

Here is a place for plugins.

After all the others: 
1. GzipBodyRequestFilter
2. ExtraTracingFilter